drivers/samsung/displays/md_series.cr

```diff
@@ -0,0 +1,500 @@
+    do_poll
+  end
+
+  # TODO: figure out what block is for
+  # def power?(**options, &block)
+  def power?(**options)
+    # options[:emit] = block unless block.nil?
+    do_send("panel_mute", [] of UInt8, **options)
+  end
+
+  # Adds mute states compatible with projectors
+  def mute(state : Bool = true)
+    power(!state)
+  end
+
+  def unmute
+    power(true)
+  end
+
+  # check software version
+  def software_version?
+    do_send("software_version")
+  end
+
+  def serial_number?
+    do_send("serial_number")
+  end
+
+  # Converts a hex encoded string into a binary string
+  def byte_to_hex(data : String) : String
+    # Removes invalid characters
+    data = data.gsub(/(0x|[^0-9A-Fa-f])*/, "")
+
+    # Ensure we have an even number of characters
+    data = '0' + data if data.size % 2 > 0
+
+    # Breaks string into an array of characters
+    output = [] of Int32
+    data.scan(/.{2}/) do |md|
+      output << md[0].to_i(16)
+    end
+
+    String.build do |io|
+      output.each do |number|
+        io.write_byte number.to_u8
+      end
+    end
+  end
+
+  # ability to send custom mdc commands via backoffice
+  def custom_mdc(command : String, value : String)
+    command = byte_to_hex(command).bytes[0].to_i
+    data = byte_to_hex(value).bytes
+    do_send(command, data)
+  end
+
+  def set_timer(enable : Bool = true, volume : Int32 = 0)
+    # set the time on the display
+    time_request = [] of Int32
+    t = Time.local
+    time_request << t.day
+    hour = t.hour
+    ampm = if hour > 12
+             hour = hour - 12
+             0 # pm
+           else
+             1 # am
+           end
+    time_request << hour
+    time_request << t.minute
+    time_request << t.month
+    year = t.year.to_s(16).rjust(4, '0')
+    time_request << year[0..1].to_i(16)
+    time_request << year[2..-1].to_i(16)
+    time_request << ampm
+
+    do_send("time", time_request)
+
+    state = enable ? "01" : "00"
+    vol = volume.to_s(16).rjust(2, '0')
+    # on 03:45 am enabled  off 03:30 am enabled  on-everyday  ignore manual  off-everyday  ignore manual  volume 15  input HDMI   holiday apply
+    custom_mdc(
+      "A4",
+      "03-2D-01 #{state}  03-1E-01   #{state}      01          80               01           80          #{vol}        21          01"
+    )
+  end
+
+  enum INPUTS
+    Vga           = 0x14 # pc in manual
+    Dvi           = 0x18
+    Dvi_video     = 0x1F
+    Hdmi          = 0x21
+    Hdmi_pc       = 0x22
+    Hdmi2         = 0x23
+    Hdmi2_pc      = 0x24
+    Hdmi3         = 0x31
+    Hdmi3_pc      = 0x32
+    Hdmi4         = 0x33
+    Hdmi4_pc      = 0x34
+    Display_port  = 0x25
+    Dtv           = 0x40
+    Media         = 0x60
+    Widi          = 0x61
+    Magic_info    = 0x20
+    Whiteboard    = 0x64
+  end
+
+  def switch_to(input : String, **options)
+    self[:input_stable] = false
+    self[:input_target] = input
+    do_send("input", INPUTS.parse(input).value, **options)
+  end
```

Comment:
As with the power control, there is also a standard interface now for switching and input selection.

https://github.com/PlaceOS/driver/blob/master/src/placeos-driver/interface/switchable.cr

Reply:
Display will use the InputSelection interface within there.

Reply:
made this commit https://github.com/PlaceOS/drivers/pull/19/commits/cfd986bc02778c27a3cb4a76c3b9d86e8ee6f4f9